### PR TITLE
Show progress in non verbose executions

### PR DESCRIPTION
**Pull request by vinzenz**
Previously when leapp was executed not much was visible during the
upgrade except for the DNF output. This patch now introduces some
changes to the execution that shows what phase starts and which
actor gets executed.

Example output:
```
[root@localhost leapp]# leapp preupgrade
==> Processing phase `configuration_phase`
====> * ipu_workflow_config
        IPU workflow config actor
==> Processing phase `FactsCollection`
====> * scanmemory
        Scan Memory of the machine.
====> * tcp_wrappers_config_read
        Parse tcp_wrappers configuration files /etc/hosts.{allow,deny}.
====> * scan_kernel_cmdline
        No documentation has been provided for the scan_kernel_cmdline actor.
====> * storage_scanner
        Provides data about storage settings.
====> * system_facts
        Provides data about many facts from system.
====> * udevadm_info
        Produces data exported by the "udevadm info" command.
====> * scan_subscription_manager_info
        Scans the current system for subscription manager information
====> * transaction_workarounds
        Provides additional RPM transaction tasks based on bundled RPM packages.
====> * rpm_scanner
        Provides data about installed RPM Packages.
Loaded plugins: product-id, subscription-manager
====> * network_manager_read_config
        Provides data about NetworkManager configuration.
====> * repository_mapping
        Produces message containing repository mapping based on provided file.
====> * sssd_facts
        Check SSSD configuration for changes in RHEL8 and report them in model.
====> * root_scanner
        Scan the system root directory and produce a message containing
====> * pam_modules_scanner
        Scan the pam directory for services and modules used in them
====> * pci_devices_scanner
        Provides data about existing PCI Devices.
```

Signed-off-by: Vinzenz Feenstra <vfeenstr@redhat.com>

### CODE REVIEW

File: leapp/utils/output.py [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
 import sys
 from contextlib import contextmanager
 
-from leapp.models import ErrorModel
 from leapp.exceptions import LeappRuntimeError
+from leapp.models import ErrorModel
+
+
+def _is_verbose():
+    """Redefinition of is_verbose from leapp.libraries.stdlib.config because it leads to import errors"""
```
Comment (Rezney) [inline leapp/utils/output.py]: Could you be please more specific about the import Errors?

I got:

```python
Traceback (most recent call last):
  File "/usr/bin/snactor", line 9, in <module>
    load_entry_point('leapp==0.9.0', 'console_scripts', 'snactor')()
  File "/usr/lib/python2.7/site-packages/leapp/snactor/__init__.py", line 77, in main
    load_commands()
  File "/usr/lib/python2.7/site-packages/leapp/snactor/__init__.py", line 19, in load_commands
    _load_commands_from(commands.__file__)
  File "/usr/lib/python2.7/site-packages/leapp/snactor/__init__.py", line 33, in _load_commands_from
    mod = importer.find_module(name).load_module(name)
  File "/usr/lib64/python2.7/pkgutil.py", line 246, in load_module
    mod = imp.load_module(fullname, self.file, self.filename, self.etc)
  File "/usr/lib/python2.7/site-packages/leapp/snactor/commands/discover.py", line 8, in <module>
    from leapp.repository.scan import find_and_scan_repositories
  File "/usr/lib/python2.7/site-packages/leapp/repository/__init__.py", line 9, in <module>
    from leapp.repository.actor_definition import ActorDefinition
  File "/usr/lib/python2.7/site-packages/leapp/repository/actor_definition.py", line 10, in <module>
    from leapp.actors import get_actor_metadata, get_actors
  File "/usr/lib/python2.7/site-packages/leapp/actors/__init__.py", line 16, in <module>
    from leapp.workflows.api import WorkflowAPI
  File "/usr/lib/python2.7/site-packages/leapp/workflows/__init__.py", line 16, in <module>
    from leapp.utils.output import display_status_current_phase, display_status_current_actor
  File "/usr/lib/python2.7/site-packages/leapp/utils/output.py", line 8, in <module>
    from leapp.libraries.stdlib.config import is_verbose
  File "/usr/lib/python2.7/site-packages/leapp/libraries/stdlib/__init__.py", line 14, in <module>
    from leapp.libraries.stdlib import api
  File "/usr/lib/python2.7/site-packages/leapp/libraries/stdlib/api.py", line 9, in <module>
    from leapp.actors import Actor
ImportError: cannot import name Actor
```
when running "snactor discover"...

Comment (Rezney) [reply leapp/utils/output.py:13]: @vinzenz any update on this?

**Comment by vinzenz** [comment]
leapp-ci build

File: leapp/utils/output.py (hunk 1 above)
Comment (Rezney) [reply leapp/utils/output.py:13]: As discussed offline, this was most probably related to the old build. The new one was stuck.

File: leapp/utils/output.py [hunk 2]
```diff
@@ -1,10 +1,17 @@
...
+    return os.getenv('LEAPP_DEBUG', '0') == '1' or os.getenv('LEAPP_VERBOSE', '0') == '1'
```
Comment (drehak) [inline leapp/utils/output.py:14]: You don't need to check both variables. When `LEAPP_DEBUG` is set to 1, `LEAPP_VERBOSE` gets set too.
```suggestion
    return os.getenv('LEAPP_VERBOSE', '0') == '1'
```

**Comment by Rezney** [comment]
Sometimes, the output seems to be pretty mixed into the other output:

```
  unbound-libs-1.7.3-8.el8.x86_64                                               
  xkeyboard-config-2.24-3.el8.noarch                                            

Complete!
==> Processing phase `TargetTransactionCheck`
====> * report_set_target_release
        Reports that a release will be set in the subscription-manager after the upgrade.
====> * dnf_transaction_check
        This actor tries to solve the RPM transaction to verify the all package dependencies can be successfully resolved.
Last metadata expiration check: 0:03:41 ago on Thu Apr  2 10:23:51 2020.

```

I understand we may not be in control of this, but what about if add something in front of the arrows to distinguish those are really different logs?

Something like:

```
  unbound-libs-1.7.3-8.el8.x86_64                                               
  xkeyboard-config-2.24-3.el8.noarch                                            

Complete!
Leapp ==> Processing phase `TargetTransactionCheck`
Leapp ====> * report_set_target_release
        Reports that a release will be set in the subscription-manager after the upgrade.
Leapp ====> * dnf_transaction_check
        This actor tries to solve the RPM transaction to verify the all package dependencies can be successfully resolved.
Last metadata expiration check: 0:03:41 ago on Thu Apr  2 10:23:51 2020.
```

what do you think @pirat89 @shaded-enmity ?

**Comment by vinzenz** [comment]
Actually that's part of the example output above. See RPM Scanner.

While I can see that you might want that, I feel that to be quite convoluting the output to prefix it with a string like this. 
We rather should ask for feedback from some UI guys if we really want some feedback.
The main thing this PR is handling is that it actually shows progress of leapp itself as well. So people can see that something is happening and what it is actually doing right now.

File: leapp/utils/output.py (hunk 2 above)
Comment (vinzenz) [reply leapp/utils/output.py:14]: @drehak it's a copy and paste of the original implementation

Comment (drehak) [reply leapp/utils/output.py:14]: I see. Still, debug also sets verbose in cli code now.

**Comment by Rezney** [comment]
@vinzenz ok, fair enough. Looks good to me then.

**Comment by Rezney** [comment]
@drehak what do you think?

**Comment by drehak** [comment]
@Rezney Yeah, looks good.
The displayed actor descriptions differ a lot and gramatically, some of them quite stand out. We could "normalize" those a bit when we have time, but the strings are in in leapp-repository anyway, so not a problem here.

Just address the requested change above and we can merge. :)

**Comment by drehak** [comment]
Though there could also be something like
`==> Loading actors`
at the start, since it seemingly hangs there for a few seconds.

**Comment by vinzenz** [comment]
e2e tests

**Pull request merged**